pkg/apis/provisioning/v1alpha3/provisioner_validation.go

```diff
@@ -103,6 +106,23 @@ func (c *Cluster) validate() (errs *apis.FieldError) {
 	return errs
 }
 
+// HasTLSEndpoint returns whether or not the endpoint requires TLS
+// (and, most likely, depending on cloud provider, a CABundle).
+// Conservatively it assumes TLS unless the endpoint has been
+// explicitly configured with a non-TLS scheme, like 'http'.
+func (c *Cluster) HasTLSEndpoint() bool {
+	endpoint, err := c.parseEndpoint()
+	return err != nil || endpoint.Scheme != "http"
+}
+
+// validate performs cloud-provider-agnostic validations (only).
+func (c *Cluster) validate() (errs *apis.FieldError) {
+	if c == nil {
+		return errs.Also(apis.ErrMissingField())
```

Comment:
Should we specify the field here?

Reply:
I see we have `.ViaField("cluster"),` above. Does this print cleanly as is?

Reply:
> Should we specify the field here?

That seems already to work:

```
error: error validating "STDIN": error validating data: ValidationError(Provisioner.spec): missing required field "cluster" in sh.karpenter.v1alpha3.Provisioner.spec; if you choose to ignore these errors, turn validation off with --validate=false
```

Regarding the ViaField question, all that code is going to change, per our offline conversation.